jerry-core/ecma/builtin-objects/ecma-builtin-number-prototype.c

```diff
@@ -472,86 +554,13 @@ ecma_builtin_number_prototype_object_to_fixed (ecma_value_t this_arg, /**< this
             *p++ = '-';
           }
 
-          int8_t digit = 0;
-          uint64_t s = 1;
-
-          /* Calculate the magnitude of the number. This is used to get the digits from left to right. */
-          while (s <= digits)
-          {
-            s *= 10;
-          }
-
-          if (exponent <= 0)
-          {
-            /* Add leading zeros. */
-            *p++ = '0';
-
-            if (frac_digits != 0)
-            {
-              *p++ = '.';
-            }
-
-            for (int i = 0; i < -exponent && i < frac_digits; i++)
-            {
-              *p++ = '0';
-            }
-
-            /* Add significant digits. */
-            for (int i = -exponent; i < frac_digits; i++)
-            {
-              digit = 0;
-              s /= 10;
-
-              while (digits >= s && s > 0)
-              {
-                digits -= s;
-                digit++;
-              }
-
-              *p = (lit_utf8_byte_t) ((lit_utf8_byte_t) digit + '0');
-              p++;
-            }
-          }
-          else
-          {
-            /* Add significant digits. */
-            for (int i = 0; i < exponent; i++)
-            {
-              digit = 0;
-              s /= 10;
-
-              while (digits >= s && s > 0)
-              {
-                digits -= s;
-                digit++;
-              }
-
-              *p = (lit_utf8_byte_t) ((lit_utf8_byte_t) digit + '0');
-              p++;
-            }
-
-            /* Add the decimal point after whole part. */
-            if (frac_digits != 0)
-            {
-              *p++ = '.';
-            }
-
-            /* Add neccessary fracion digits. */
-            for (int i = 0; i < frac_digits; i++)
-            {
-              digit = 0;
-              s /= 10;
-
-              while (digits >= s && s > 0)
-              {
-                digits -= s;
-                digit++;
-              }
-
-              *p = (lit_utf8_byte_t) ((lit_utf8_byte_t) digit + '0');
-              p++;
-            }
-          }
+          p += ecma_builtin_number_prototype_helper_to_string (digits,
+                                                               num_digits,
+                                                               exponent,
+                                                               p,
+                                                               (exponent > 0)
```

Comment:
I'd move this out of the call like you did later in the code. https://github.com/Samsung/jerryscript/pull/1118/files#diff-b4e661ee3c351fceb291f9b449c40c1aR876